src/Compilers/Core/Portable/Compilation/Expression.cs

```diff
@@ -0,0 +1,365 @@
+
+        public SyntaxNode Syntax { get; private set; }
+
+        public OperationKind Kind => OperationKind.ConditionalChoice;
+
+        public object ConstantValue => null;
+    }
+
+    public class Assignment : IExpressionStatement
+    {
+        AssignmentExpression assignment;
+
+        public Assignment(IReference target, IExpression value, SyntaxNode syntax)
+        {
+            this.assignment = new AssignmentExpression(target, value, syntax);
+            this.Syntax = syntax;
+        }
+
+        public SyntaxNode Syntax { get; set; }
+
+        public OperationKind Kind
+        {
+            get { return OperationKind.ExpressionStatement; }
+        }
+
+        public IExpression Expression
+        {
+            get { return this.assignment; }
+        }
+
+        class AssignmentExpression : IAssignment
+        {
+            public AssignmentExpression(IReference target, IExpression value, SyntaxNode syntax)
+            {
+                this.Value = value;
+                this.Target = target;
+                this.Syntax = syntax;
+            }
+
+            public IReference Target { get; set; }
+
+            public IExpression Value { get; set; }
+
+            public SyntaxNode Syntax { get; set; }
+
+            public ITypeSymbol ResultType
+            {
+                get { return this.Target.ResultType; }
+            }
+
+            public OperationKind Kind
+            {
+                get { return OperationKind.Assignment; }
+            }
+
+            public object ConstantValue
+            {
+                get { return null; }
+            }
+        }
+    }
+
+    public class CompoundAssignment : IExpressionStatement
+    {
+        CompoundAssignmentExpression compoundAssignment;
+
+        public CompoundAssignment(IReference target, IExpression value, BinaryOperationKind binaryKind, IMethodSymbol operatorMethod, SyntaxNode syntax)
+        {
+            this.compoundAssignment = new CompoundAssignmentExpression(target, value, binaryKind, operatorMethod, syntax);
+            this.Syntax = syntax;
+        }
+
+        public SyntaxNode Syntax { get; set; }
+
+        public OperationKind Kind
+        {
+            get { return OperationKind.ExpressionStatement; }
+        }
+
+        public IExpression Expression
+        {
+            get { return this.compoundAssignment; }
+        }
+
+        class CompoundAssignmentExpression : ICompoundAssignment
+        {
+            public CompoundAssignmentExpression(IReference target, IExpression value, BinaryOperationKind binaryKind, IMethodSymbol operatorMethod, SyntaxNode syntax)
+            {
+                this.Target = target;
+                this.Value = value;
+                this.BinaryKind = binaryKind;
+                this.Operator = operatorMethod;
+                this.Syntax = syntax;
+            }
+
+            public IReference Target { get; set; }
```

Comment:
I was partway through cleaning up this file. There's a bunch of litter that needs to go.